packages/mds-cache/index.ts

```diff
@@ -340,21 +340,38 @@ async function readDevices(device_ids: UUID[]) {
 async function readDeviceStatus(device_id: UUID) {
   let event: VehicleEvent
   let device: Device
+  const cachedInfo = []
   try {
     event = await readEvent(device_id)
+    cachedInfo.push(event)
+  } catch (err) {
+    if (err.name !== 'NotFoundError') {
+      throw err
+    }
+    log.info('Missing vehicle event', 'device_id', device_id)
+  }
+  try {
     device = await readDevice(device_id)
-  } catch {
-    return null
+    cachedInfo.push(device)
+  } catch (err) {
+    if (err.name !== 'NotFoundError') {
+      throw err
+    }
+    log.info('Missing vehicle device', 'device_id', device_id)
   }
+
   const deviceStatusMap: { [device_id: string]: CachedItem | {} } = {}
-  const all = [device, event]
-  all.map(item => {
+  cachedInfo.map(item => {
     deviceStatusMap[item.device_id] = deviceStatusMap[item.device_id] || {}
     Object.assign(deviceStatusMap[item.device_id], item)
   })
-  const values = Object.values(deviceStatusMap)
-
-  return values.filter((item: any) => item.telemetry)[0]
+  const statuses = Object.values(deviceStatusMap)
+  const statusesWithTelemetry = statuses.filter((status: any) => status.telemetry)
```

Comment:
It might be more semantic to use `.find()` here, since we're always indexing into the first element of the filtered array.

Reply:
Nice, agreed.